registry/storage/registry.go

```diff
@@ -41,18 +78,22 @@ func NewRegistryWithDriver(ctx context.Context, driver storagedriver.StorageDriv
 		statter: statter,
 	}
 
-	return &registry{
+	registry := &registry{
 		blobStore: bs,
 		blobServer: &blobServer{
-			driver:   driver,
-			statter:  statter,
-			pathFn:   bs.path,
-			redirect: redirect,
+			driver:  driver,
+			statter: statter,
+			pathFn:  bs.path,
 		},
-		blobDescriptorCacheProvider: blobDescriptorCacheProvider,
-		deleteEnabled:               deleteEnabled,
-		resumableDigestEnabled:      !isCache,
+		statter:                statter,
+		resumableDigestEnabled: true,
 	}
+
+	for _, option := range options {
+		option(registry)
```

Comment:
We are losing any error returns from applying an option here.  I think they should be propagated out of NewRegistry